### PR TITLE
Change how plugins work

### PR DESCRIPTION
**Pull request by chuckha**
Instead of creating our own schema for a plugin we will use annotations
to get metadata to sonobuoy instead. We can then define full kubernetes
resources as plugins.

The resources we define are templates. The idea is to eventually move
these templates over to ksonnet and have a nice ksonnet library for the
shared pieces between plugins.

Remaining work:

* Whatever configuration can be passed into the template
should be allowed to be passed into the template (`RESULTS_DIR`).
* Add a test or two to the plugin.Interface implementations

Fixes: #96 

Signed-off-by: Chuck Ha <chuck@heptio.com>

### CODE REVIEW

**Comment by chuckha** [comment]
[201711071651_sonobuoy_d2e178db-0eeb-4c35-9001-f3d84be72fc2.tar.gz](https://github.com/heptio/sonobuoy/files/1450899/201711071651_sonobuoy_d2e178db-0eeb-4c35-9001-f3d84be72fc2.tar.gz)

Here is a sample run using this code.

File: pkg/plugin/loader/loader.go [hunk 1]
```diff
@@ -97,107 +96,54 @@ func loadPlugin(namespace string, dfn plugin.Definition, masterAddress string) (
...
-		if err != nil {
-			return plugins, err
-		}
-
-		pluginDef, err := loaderFn(y)
+		pluginTemplate, err := ioutil.ReadFile(fullPath)
 		if err != nil {
-			logrus.Warningf("Error unmarshalling bytes at %v: %v", fullPath, err)
-			continue
+			return pluginDfns, err
```
Comment (timothysc) [inline pkg/plugin/loader/loader.go]: why not return nil,err on failure here?

Comment (timothysc) [reply pkg/plugin/loader/loader.go]: I realize it's nil, but it's less explicit when reading.

File: examples/quickstart.yaml [hunk 2]
```diff
@@ -261,7 +290,7 @@ spec:
       valueFrom:
         fieldRef:
           fieldPath: status.podIP
-    image: gcr.io/heptio-images/sonobuoy:master
+    image: gcr.io/heptio-images/sonobuoy:latest
```
Comment (timothysc) [inline examples/quickstart.yaml]: should be master

File: pkg/worker/config.go [hunk 3]
```diff
@@ -47,6 +47,7 @@ func LoadConfig() (*plugin.WorkerConfig, error) {
 	viper.BindEnv("masterurl", "MASTER_URL")
 	viper.BindEnv("nodename", "NODE_NAME")
 	viper.BindEnv("resultsdir", "RESULTS_DIR")
+	viper.BindEnv("resulttype", "RESULT_TYPE")
```
Comment (timothysc) [inline pkg/worker/config.go]: Look into this one.

Comment (chuckha) [reply pkg/worker/config.go]: We do need this. I am providing these values to the worker via Env Vars in the plugin template. This allows me to bypass the ConfigMap entirely.

**Pull request merged**